### PR TITLE
Fix serializer replacing non emotion class names

### LINKED ISSUE #706: jest-emotion 9.2.0 replacing non-emotion class names
- `emotion` version: 9.2.1 **(jest-emotion: 9.2.0)**
- `react` version: 16.4.0

Relevant code.

```javascript
import React from "react";
import { css, cx } from "emotion";
import renderer from "react-test-renderer";

const classes = cx(
  "net-42",
  "net",
  css`
    color: darkorchid;
  `
);

it("renders with correct styles", () => {
  expect(renderer.create(<div className={classes} />)).toMatchSnapshot();
});
```

What you did:

Upgraded jest-emotion from 9.1.3 to 9.2.0.

What happened:

Serialized snapshot tests failed due to incorrect classes being applied.

```css
- Snapshot
+ Received

<div
-  className="net-42 net emotion-0"
+  className="emotion-0 net emotion-1"
/>
```

Reproduction:

https://github.com/wKovacs64/jest-emotion-testing

Problem description:

CSS classes that match the [`/e[a-zA-Z0-9-]+[0-9]+/`](https://github.com/emotion-js/emotion/blob/ee71446ac21861f938f51b5696db956d75e8ce87/packages/jest-emotion/src/replace-class-names.js#L8) regex (e.g. `net-42`) are being treated as `emotion` classes.

Suggested solution:

I think this came from https://github.com/emotion-js/emotion/pull/641 and the regex just needs updating?

### PR DESCRIPTION
**Pull request by mitchellhamilton**
**What**:
Fix serializer replacing non emotion class names

**Why**:
Closes #706 

**How**:

Make the regex used to replace the component selector class names more strict.

**Checklist**:

- [ ] Documentation N/A
- [x] Tests
- [x] Code complete

### CODE REVIEW

File: packages/jest-emotion/src/replace-class-names.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ function defaultClassNameReplacer(className, index) {
 
 export type ClassNameReplacer = (className: string, index: number) => string
 
-const componentSelectorClassNamePattern = /e[a-zA-Z0-9-]+[0-9]+/
+const componentSelectorClassNamePattern = /^e[a-zA-Z0-9]+[0-9]+$/
```
Comment (Ailrun) [inline packages/jest-emotion/src/replace-class-names.js]: Where is the `'-'`? I mean, is it OK to remove `'-'`?
This pattern does not match with `emotion-0`....

Comment (mitchellhamilton) [reply packages/jest-emotion/src/replace-class-names.js]: It's not meant to match `emotion-0`.

It's meant to match class names like this `e17ivhiq0`.

Component selector class names used to be like `css-17ivhiq0` so this included `-` and I just forgot to remove it when I changed the shape of the component selector class names

Comment (Ailrun) [reply packages/jest-emotion/src/replace-class-names.js]: Thank you for your detailed answer! :)

Comment (Ailrun) [reply packages/jest-emotion/src/replace-class-names.js]: @mitchellhamilton Wa...it, this also does not match with `e17ivhiq0`, does it?
It matches with `e17` or `eifjwe5239`, but does not with `e17ivhiq0`.

Comment (Ailrun) [reply packages/jest-emotion/src/replace-class-names.js]: Never mind. I'm the fool.

**Pull request merged**